### ISSUE TITLE
Physical buttons don't store shutterposition

### ISSUE THREAD

**Issue by epandi** [arendst/Tasmota#15838]
### PROBLEM DESCRIPTION
When using a physical button, the shutter position is not stored to flash. When using the webbuttons this does happen.
This means that a powerloss or physicial replugging of the power cord in the wall socket resets the shutter position.
After a powerloss i now have to press the open button (in opened state, while tasmota reports the shutter as being closed) so the switch is putting power in the open relay for 30 seconds (eventually doing nothing to the blinds other then getting the slider back to opened state). Afterwards I can use the close button again to really close the shutter.

In version 11.0 this wasn't an issue.

### REQUESTED INFORMATION

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): DS-102 2 Gang
- [X] Tasmota binary firmware version number used: 12.0.1
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: Web update
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`:

```lua
  Configuration output here:
15:16:05.459 MQT: stat/fleurblinds/RESULT = {"NAME":"DS-102 2 Gang","GPIO":[576,321,0,32,225,33,0,0,0,224,320,1,0,0],"FLAG":0,"BASE":18}
15:16:05.674 MQT: stat/fleurblinds/RESULT = {"Module":{"0":"DS-102 2 Gang"}}
15:16:05.928 MQT: stat/fleurblinds/RESULT = {"GPIO0":{"576":"LedLink_i"},"GPIO1":{"321":"Led_i2"},"GPIO2":{"0":"None"},"GPIO3":{"32":"Button1"},"GPIO4":{"225":"Relay2"},"GPIO5":{"33":"Button2"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"224":"Relay1"},"GPIO14":{"320":"Led_i1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```
- [X] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:

```lua
  Rules output here:
15:16:22.184 MQT: stat/fleurblinds/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
15:16:22.400 MQT: stat/fleurblinds/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":67,"Free":444,"Rules":"ON Clock#Timer=2 DO Rule1 0 ENDON ON Clock#Timer=3 DO Rule1 1 ENDON"}}
15:16:22.649 MQT: stat/fleurblinds/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":108,"Free":403,"Rules":"ON System#Boot DO Shutterposition1 ENDON ON Shutter1#Position DO Publish stat/%topic%/SHUTTER1 %value% ENDON"}}
```
- [X] Provide the output of this command: `Status 0`:

```lua
  STATUS 0 output here:
15:16:38.607 MQT: stat/fleurblinds/STATUS = {"Status":{"Module":0,"DeviceName":"FleurBlinds","FriendlyName":["FleurBlinds","FleurBlinds"],"Topic":"fleurblinds","ButtonTopic":"","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":0,"SaveState":1,"SwitchTopic":"0","SwitchMode":[5,5,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
15:16:38.639 MQT: stat/fleurblinds/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"External System","Uptime":"0T00:25:43","StartupUTC":"2022-06-21T12:50:55","Sleep":50,"CfgHolder":4617,"BootCount":811,"BCResetTime":"2020-05-01T08:02:39","SaveCount":2590,"SaveAddress":"F7000"}}
15:16:38.676 MQT: stat/fleurblinds/STATUS2 = {"StatusFWR":{"Version":"12.0.1(tasmota)","BuildDateTime":"2022-06-17T14:38:38","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"405/699"}}
15:16:38.696 MQT: stat/fleurblinds/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["XXXXXXXXXX","Wurmhole"],"TelePeriod":300,"Resolution":"558180C0","SetOption":["0000C809","1405C80001000600003C5A0A000000000000","400000C9","00006000","00004000","00000000"]}}
15:16:38.730 MQT: stat/fleurblinds/STATUS4 = {"StatusMEM":{"ProgramSize":618,"Free":384,"Heap":24,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"1540C8","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","14000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
15:16:38.766 MQT: stat/fleurblinds/STATUS5 = {"StatusNET":{"Hostname":"fleurblinds-2933","IPAddress":"192.168.1.32","Gateway":"192.168.0.1","Subnetmask":"255.255.252.0","DNSServer1":"8.8.4.4","DNSServer2":"0.0.0.0","Mac":"EC:FA:BC:4B:0B:75","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
15:16:38.790 MQT: stat/fleurblinds/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.0.222","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_4B0B75","MqttUser":"sonoff","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
15:16:38.814 MQT: stat/fleurblinds/STATUS7 = {"StatusTIM":{"UTC":"2022-06-21T13:16:38","Local":"2022-06-21T15:16:38","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":99,"Sunrise":"05:46","Sunset":"21:57"}}
15:16:38.833 MQT: stat/fleurblinds/STATUS10 = {"StatusSNS":{"Time":"2022-06-21T15:16:38","Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0}}}
15:16:38.847 MQT: stat/fleurblinds/STATUS11 = {"StatusSTS":{"Time":"2022-06-21T15:16:38","Uptime":"0T00:25:43","UptimeSec":1543,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"XXXXXXXXXX","BSSId":"28:6C:07:98:E3:95","Channel":6,"Mode":"11n","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:07"}}}
15:16:38.879 MQT: stat/fleurblinds/STATUS13 = {"StatusSHT":{"SHT0":{"Relay1":1,"Relay2":2,"Open":330,"Close":330,"50perc":50,"Delay":0,"Opt":"0000","Calib":[300,500,700,900,1000],"Mode":"0"}}}
```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:

```lua
  Console output here:
00:00:02.167 WIF: Attempting connection...
00:00:03.166 WIF: Attempting connection...
00:00:04.519 WIF: Connecting to AP1 XXXXXXXXXX Channel 6 BSSId 28:6C:07:98:E3:95 in mode 11n as fleurblinds-2933...
00:00:04.520 WIF: Attempting connection...
00:00:05.803 WIF: Connected
00:00:06.054 HTP: Web server active on fleurblinds-2933 with IP address 192.168.1.32
00:00:06.535 QPC: Reset
00:00:06.581 RTC: UTC 2022-06-21T12:49:15, DST 2022-03-27T02:00:00, STD 2022-10-30T03:00:00
14:49:15.000 RTC: Synced by NTP
14:49:16.039 MQT: Attempting connection...
14:49:16.058 MQT: Connected
14:49:16.061 MQT: tele/fleurblinds/LWT = Online (retained)
14:49:16.065 MQT: cmnd/fleurblinds/POWER = 
14:49:16.068 MQT: Subscribe to cmnd/fleurblinds/#
14:49:16.072 MQT: Subscribe to cmnd/tasmotas/#
14:49:16.076 MQT: Subscribe to cmnd/DVES_4B0B75_fb/#
14:49:16.081 MQT: tele/fleurblinds/INFO1 = {"Info1":{"Module":"DS-102 2 Gang","Version":"12.0.1(tasmota)","FallbackTopic":"cmnd/DVES_4B0B75_fb/","GroupTopic":"cmnd/tasmotas/"}}
14:49:16.092 MQT: tele/fleurblinds/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"fleurblinds-2933","IPAddress":"192.168.1.32"}}
14:49:16.103 MQT: tele/fleurblinds/INFO3 = {"Info3":{"RestartReason":"External System","BootCount":810}}
14:49:16.112 MQT: stat/fleurblinds/RESULT = {"POWER1":"OFF"}
14:49:16.117 MQT: stat/fleurblinds/POWER1 = OFF
14:49:16.122 MQT: stat/fleurblinds/RESULT = {"POWER2":"OFF"}
14:49:16.127 MQT: stat/fleurblinds/POWER2 = OFF
14:49:17.436 APP: Boot Count 810
14:49:17.608 CFG: Saved to flash at F8, Count 2581, Bytes 4096
14:49:20.426 MQT: tele/fleurblinds/STATE = {"Time":"2022-06-21T14:49:20","Uptime":"0T00:00:13","UptimeSec":13,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":21,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"XXXXXXXXXX","BSSId":"28:6C:07:98:E3:95","Channel":6,"Mode":"11n","RSSI":100,"Signal":-50,"LinkCount":1,"Downtime":"0T00:00:07"}}
14:49:20.458 MQT: tele/fleurblinds/SENSOR = {"Time":"2022-06-21T14:49:20","Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}
14:49:30.388 CMD: shutterposition
14:49:30.390 SRC: WebConsole from 192.168.0.129
14:49:30.393 CMD: Grp 0, Cmd 'SHUTTERPOSITION', Idx 1, Len 0, Pld -99, Data ''
14:49:30.401 MQT: stat/fleurblinds/RESULT = {"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}
14:49:30.407 MQT: stat/fleurblinds/RESULT = {"Position1":100}
14:49:37.891 HTP: Main Menu
14:49:57.050 HTP: Console
14:50:04.393 APP: Button1 multi-press 1
14:50:04.971 SRC: Button
14:50:04.974 SRC: Shutter
14:50:04.979 MQT: stat/fleurblinds/RESULT = {"POWER1":"OFF"}
14:50:04.983 MQT: stat/fleurblinds/POWER1 = OFF
14:50:04.987 MQT: stat/fleurblinds/RESULT = {"POWER1":"OFF"}
14:50:04.990 MQT: stat/fleurblinds/POWER1 = OFF
14:50:04.994 SHT: Shtr1 Real 660000, Start 660000, Stop 660000, Dir 1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
14:50:04.997 MQT: stat/fleurblinds/RESULT = {"Shutter1":{"Position":100,"Direction":1,"Target":100,"Tilt":0}}
14:50:04.000 SHT: Stop 1 Mode 0 time 0
14:50:05.501 SHT: Shtr1 Real 660000, Start 660000, Stop 660000, Dir 0, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 0
14:50:05.506 MQT: stat/fleurblinds/SHUTTER1 = 100
14:50:05.511 MQT: stat/fleurblinds/RESULT = {"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}
14:50:06.573 APP: Button2 multi-press 1
14:50:07.190 SRC: Button
14:50:07.196 MQT: stat/fleurblinds/RESULT = {"POWER2":"ON"}
14:50:07.201 MQT: stat/fleurblinds/POWER2 = ON
14:50:07.207 SHT: Shtr1 Real 660000, Start 660000, Stop 0, Dir -1, Delay 0, Rtc 0.05 [s], Freq 0, PWM 0, Tilt 0
14:50:07.212 MQT: stat/fleurblinds/RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":0,"Tilt":0}}
14:50:39.456 SHT: Shtr1 Real 15000, Start 660000, Stop 0, Dir -1, Delay 0, Rtc 32.30 [s], Freq 0, PWM 0, Tilt 0
14:50:39.461 MQT: stat/fleurblinds/RESULT = {"Shutter1":{"Position":2,"Direction":-1,"Target":0,"Tilt":0}}
14:50:40.215 SHT: Stop 1 Mode 0 time 661
14:50:40.217 SRC: Shutter
14:50:40.223 MQT: stat/fleurblinds/RESULT = {"POWER2":"OFF"}
14:50:40.228 MQT: stat/fleurblinds/POWER2 = OFF
14:50:40.731 SHT: Shtr1 Real 0, Start 0, Stop 0, Dir 0, Delay 0, Rtc 33.05 [s], Freq 0, PWM 0, Tilt 0
14:50:40.735 MQT: stat/fleurblinds/SHUTTER1 = 0
14:50:40.740 MQT: stat/fleurblinds/RESULT = {"Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0}}

00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F8, Count 2581
00:00:00.056 QPC: Count 1
00:00:00.058 CFG: CR 405/699, Busy 0
00:00:00.062 ROT: Mode 1
00:00:00.064 SHT: Shtr1 min realpos_chg: 2000, min tilt_chg 2
00:00:00.065 SHT: Shtr1 Init. Pos 660000, Inv 0, Locked 0, Endstop enab 0, webButt inv 0, Motordel: 0
00:00:00.066 SRC: Restart
00:00:00.069 Project tasmota - FleurBlinds Version 12.0.1(tasmota)-2_7_4_9(2022-06-17T14:38:38)
00:00:00.223 WIF: Attempting connection...
00:00:01.169 WIF: Network (re)scan started...
00:00:01.170 WIF: Attempting connection...
00:00:02.169 WIF: Attempting connection...
00:00:03.171 WIF: Attempting connection...
00:00:04.473 WIF: Connecting to AP1 XXXXXXXXXX Channel 6 BSSId 28:6C:07:98:E3:95 in mode 11n as fleurblinds-2933...
00:00:04.475 WIF: Attempting connection...
00:00:05.756 WIF: Connected
00:00:06.009 HTP: Web server active on fleurblinds-2933 with IP address 192.168.1.32
00:00:06.536 QPC: Reset
00:00:06.581 RTC: UTC 2022-06-21T12:51:03, DST 2022-03-27T02:00:00, STD 2022-10-30T03:00:00
14:51:03.001 RTC: Synced by NTP
14:51:04.039 MQT: Attempting connection...
14:51:04.055 MQT: Connected
14:51:04.058 MQT: tele/fleurblinds/LWT = Online (retained)
14:51:04.062 MQT: cmnd/fleurblinds/POWER = 
14:51:04.065 MQT: Subscribe to cmnd/fleurblinds/#
14:51:04.067 MQT: Subscribe to cmnd/tasmotas/#
14:51:04.071 MQT: Subscribe to cmnd/DVES_4B0B75_fb/#
14:51:04.080 MQT: tele/fleurblinds/INFO1 = {"Info1":{"Module":"DS-102 2 Gang","Version":"12.0.1(tasmota)","FallbackTopic":"cmnd/DVES_4B0B75_fb/","GroupTopic":"cmnd/tasmotas/"}}
14:51:04.091 MQT: tele/fleurblinds/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"fleurblinds-2933","IPAddress":"192.168.1.32"}}
14:51:04.102 MQT: tele/fleurblinds/INFO3 = {"Info3":{"RestartReason":"External System","BootCount":811}}
14:51:04.111 MQT: stat/fleurblinds/RESULT = {"POWER1":"OFF"}
14:51:04.116 MQT: stat/fleurblinds/POWER1 = OFF
14:51:04.121 MQT: stat/fleurblinds/RESULT = {"POWER2":"OFF"}
14:51:04.124 MQT: stat/fleurblinds/POWER2 = OFF
14:51:05.483 APP: Boot Count 811
14:51:05.607 CFG: Saved to flash at F7, Count 2582, Bytes 4096
14:51:08.467 MQT: tele/fleurblinds/STATE = {"Time":"2022-06-21T14:51:08","Uptime":"0T00:00:13","UptimeSec":13,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":22,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"XXXXXXXXXX","BSSId":"28:6C:07:98:E3:95","Channel":6,"Mode":"11n","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:07"}}
14:51:08.498 MQT: tele/fleurblinds/SENSOR = {"Time":"2022-06-21T14:51:08","Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}
14:51:12.649 HTP: Main Menu
14:51:23.347 HTP: Console
```

### TO REPRODUCE
Hit the physical up/down button on the switch.
"CFG: Saved to flash at F6, Count 2583, Bytes 4096" is not triggered.
This does happen when using the web button.
![savetoflash](https://user-images.githubusercontent.com/958386/174810172-49cebd2a-256e-468e-a058-05defd56f809.png)

### EXPECTED BEHAVIOUR
Position saved to flash so the position isn't reset during a powerloss or replug of the power cord.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by epandi** [comment]
Update
After updating to 12.0.2.2 the shutter position is not stored to flash at all anymore. Not using the physical buttons nor using the web buttons.

In the logging I can't see a CFG: Saved to flash at C5 line.

**Comment by epandi** [comment]
Apparently Savedata got reset to 0, maybe during the upgrade or wasn't really configured correctly by me in the past. Enabling Savedata resolved the issue in v12.0.2.2.
